components/inputs/input-date.js

```diff
@@ -0,0 +1,176 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { formatDate, parseDate } from '@brightspace-ui/intl/lib/dateTime.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { inputLabelStyles } from './input-label-styles.js';
+import { inputStyles } from './input-styles.js';
+import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+function formatDateInISO(val) {
+	let month = parseInt(val.getMonth()) + 1;
+	let date = val.getDate();
+	if (month < 10) month = `0${month}`;
+	if (date < 10) date = `0${date}`;
+	return `${val.getFullYear()}-${month}-${date}`;
+}
+
+function parseISODate(val) {
+	if (!val) return null;
+	const re = /([0-9]{4})-([0-9]{2})-([0-9]{2})/;
+	const match = val.match(re);
+	if (!match || match.length !== 4) {
+		throw new Error('Invalid value: Expected format is YYYY-MM-DD');
+	}
+
+	return new Date(parseInt(match[1]), parseInt(match[2]) - 1, parseInt(match[3]));
+}
+
+class InputDate extends LocalizeStaticMixin(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean },
+			label: { type: String },
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			placeholder: { type: String },
+			value: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ inputStyles, inputLabelStyles,
+			css`
+				:host {
+					display: inline-block;
+					min-width: 7rem;
+					width: 100%;
+				}
+				:host([hidden]) {
+					display: none;
+				}
+				label {
+					display: block;
+				}
+				.d2l-input-date-container {
+					position: relative;
+				}
+				.d2l-input {
+					padding-left: 2.4rem;
+					overflow: hidden;
+					text-overflow: ellipsis;
+					white-space: nowrap;
+					-webkit-appearance: textfield;
+				}
+				:host([dir="rtl"]) .d2l-input {
+					padding-left: 0.8rem;
+					padding-right: 2.4rem;
+				}
+				.d2l-input:hover,
+				.d2l-input:focus,
+				:host([disabled]) .d2l-input {
+					padding-left: calc(2.4rem - 1px);
+					padding-right: calc(0.8rem - 1px);
+				}
+				:host([dir="rtl"]) .d2l-input:hover,
+				:host([dir="rtl"]) .d2l-input:focus {
+					padding-left: calc(0.8rem - 1px);
+					padding-right: calc(2.4rem - 1px);
```

Comment:
Agreed - should I put something up within this PR or work on it separately and switch this and `d2l-input-search` over?

Reply:
I don't feel strongly about it, but a separate PR might be nice just so we can review a smaller subset of changes, and that way it can get it right away without being blocked on this work.